src/main/java/appeng/core/features/DamagedItemDefinition.java

```diff
@@ -32,7 +34,7 @@
 {
 	private final IStackSrc source;
 
-	public DamagedItemDefinition( IStackSrc source )
+	public DamagedItemDefinition( @Nonnull IStackSrc source )
```

Comment:
If we draw a line between `private` and `public/protected` (and maybe `package private` somewhere) and not between `api` and say `do not implement`. Then yes, `Preconditions` are better, but should then apply to every `Definition`, as these are all using assert currently.